app/components/views/Offset/index.tsx

```diff
@@ -120,12 +120,13 @@ export const Offset = (props: Props) => {
   const [cost, setCost] = useState("");
   const [beneficiary, setBeneficiary] = useState("");
   const [beneficiaryAddress, setBeneficiaryAddress] = useState("");
+  const [beneficiaryAddressError, setBeneficiaryAddressError] = useState(false);
   const [retirementMessage, setRetirementMessage] = useState("");
   // for selective retirement
-  const [specificAddresses, setSpecificAddresses] = useState([""]);
-  const validSpecificAddresses = specificAddresses.filter(
-    (str) => str.length === 42 && str.startsWith("0x")
-  );
+  const [specificAddresses, setSpecificAddresses] = useState({
+    addresses: [""],
+    errors: [false],
```

Comment:
No need to persist these errors in state. We can just derive the error state at render time (components already rerender with every `onChange` event)

Cleanest would be to create a helper that returns a boolean and invoke it like this:
```
// at the input component
<input
  data-error={isValidAddress(specificAddresses[i]) })} // conditional className is also an option
 />
 
// styles.ts
export const input = css`
  &[data-error="true"]:not(:focused) {
   border: 0.2rem solid red;
  }
`;
```

I'm kind of a fan of `data` attributes because they are concise but you can also do a conditional classname using the `cx` util from emotion

Reply:
updated to use data attr